arch/arm/core/aarch32/cortex_m/mpu/arm_core_mpu.c

```diff
@@ -148,6 +148,20 @@ void z_arm_configure_dynamic_mpu_regions(struct k_thread *thread)
 	 * of the respective dynamic MPU regions to be programmed for
 	 * the given thread. The array of partitions (along with its
 	 * actual size) will be supplied to the underlying MPU driver.
+	 *
+	 * The drivers of what regions get configured are CONFIG_USERSPACE,
+	 * CONFIG_MPU_STACK_GUARD, and K_USER/supervisor threads.
+	 *
+	 * If CONFIG_USERSPACE then any Memory Domains associated with the
+	 * thread get a defined region.
```

Comment:
A memory domain is a collection of regions that the thread has access to, might be more precise to say that if the thread is a member of a memory domain, that any partitions defined within that domain get programmed to the MPU when the thread is context switched in.

Reply:
integrated some of this